### DISCUSSION TITLE
Parallelism and determinism

### DISCUSSION THREAD

**Discussion by Mapiarz** [google/or-tools#3275] [CP-SAT questions] [answered]
Hi!

In this discussion thread: https://groups.google.com/g/or-tools-discuss/c/i3TF_Szuz_k @lperron mentioned parallel search and determinism is something that you would like to add to or-tools. 

What is the status of deterministic parallel search? If not already available, is it on the roadmap at all?

Thanks

**Comment by lperron** [comment]
For cp-sat I guess. We are not working on that. Currently, the cost is too
high.

**Comment by maliberty** [comment]
How do you suggest testing a program that must use threads due to the problem size but can't a deterministic result?  It would be beneficial to have a deterministic option.

**Comment by lperron** [comment, marked as answer]
Using the main branch, you can turn on deterministic search:

interleave_search:true share_clauses:false num_workers:<number of cores on your computer>

Please be aware that it is much slower than non deterministic search.

It will be available in version 9.6
